### PR TITLE
GDAL_HTTPS_PROXY

### PR DESCRIPTION
**Pull request by drnextgis**
## What does this PR do?

This PR adds new environment variable `GDAL_HTTPS_PROXY` which is taken into account only for https URLS. If this variable is not set then `GDAL_HTTP_PROXY` affects both http and https URLs (this behavior is preserved). It is meant to overcome this issue: https://github.com/OSGeo/gdal/issues/878

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [x] All CI builds and checks have passed

### CODE REVIEW

**Comment by rouault** [comment]
Can you document the new option in the doc of CPLHTTPFetch() ?

**Comment by drnextgis** [comment]
Done.

File: gdal/port/cpl_http.cpp [hunk 1]
```diff
@@ -575,7 +575,10 @@ static void CPLHTTPEmitFetchDebug(const char* pszURL,
  * <li>POSTFIELDS=val, where val is a nul-terminated string to be passed to the server
  *                     with a POST request.</li>
  * <li>PROXY=val, to make requests go through a proxy server, where val is of the
- *                form proxy.server.com:port_number</li>
+ *                form proxy.server.com:port_number. This option affects both HTTP and HTTPS
+ *                URLs.</li>
+ * <li>HTTPS_PROXY=val, the same meaning as PROXY, but this option is taken into account only
```
Comment (rouault) [inline gdal/port/cpl_http.cpp]: Indicate the version with
```HTTPS_PROXY=val (GDAL >= 2.4),```

Comment (drnextgis) [reply gdal/port/cpl_http.cpp]: Added.

**Comment by drnextgis** [comment]
@rouault should we expect that these changes will be in 2.4.0 release and do you have an approximate date for this release?

**Comment by drnextgis** [comment]
Ok, thanks.

**Comment by jratike80** [comment]
The ticket #878 is still open but I guess it could be closed.

**Pull request merged**